rafthttp/peer.go

```diff
@@ -271,12 +264,14 @@ func (p *peer) pick(m raftpb.Message) (writec chan<- raftpb.Message, picked stri
 	// stream for a long time, only use one of the N pipelines to send MsgSnap.
 	if isMsgSnap(m) {
 		return p.pipeline.msgc, pipelineMsg
-	} else if writec, ok = p.msgAppWriter.writec(); ok && canUseMsgAppStream(m) {
```

Comment:
OK. Can you please separate this into another commit? This is actually a bug fixing. Or you might think it does not worth the effort. But we need to make sure this is test-covered.